### ISSUE TITLE
TabView: TabWidthMode set to Compact is ignored from XAML markup and for newly-added items when using data binding

### ISSUE THREAD

**Issue by Felix-Dev** [microsoft/microsoft-ui-xaml#3114]
**Describe the bug**
The exact same issues reported in #2711 and #2724 also apply to the TabView when its tab items are supplied va data-binding:

#### XAML Markup issue
```xml
<controls:TabView x:Name="DataBindingTabView" SelectedIndex="2"
                  TabItemsSource="{x:Bind ItemsSource, Mode=OneWay}"
                  TabWidthMode="Compact"
                  AddTabButtonClick="TabViewAddButtonClick">
    <controls:TabView.TabItemTemplate>
        <!-- left out here --> 
    </controls:TabView.TabItemTemplate>
</controls:TabView>
```
![image](https://user-images.githubusercontent.com/1398851/90073291-39a05e00-dcf9-11ea-9005-84af45438f70.png)

#### Adding an item to the bound data source:
![tabview-databinding-compact-issue](https://user-images.githubusercontent.com/1398851/90074208-0232b100-dcfb-11ea-885f-10ecec791db1.gif)

**Expected behavior**
All TabViewItems should be in compact mode when the TabView is loaded or when a new item is added to the TabView.ItemsSource.

**Version Info**

NuGet package version: 
WinUI 2.4 and WinUI 2.5 previews.

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| May 2020 Update (19041)            | Yes   |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**
I would like to fix this. The fix for this issue, if accepted, would also fix the bug reported in https://github.com/microsoft/microsoft-ui-xaml/issues/3104.

**Comment by chingucoding** [comment]
> The fix for this issue, if accepted, would also fix the bug reported in #3104.

So you already have a fix for this issue at hand?

**Comment by Felix-Dev** [comment]
@chingucoding Yes, my current implementation of a fix for this issue here (currently working on tests) also fixes issue #3104. Mentioned this above so that no one will invest time into specifically providing a fix for #3104 when a fix could already arrive with a PR for this issue.
